pkgs/development/compilers/rust/rustc.nix

```diff
@@ -227,6 +227,13 @@ in stdenv.mkDerivation (finalAttrs: {
 
     # Useful debugging parameter
     # export VERBOSE=1
+  '' + lib.optionalString stdenv.targetPlatform.isDarwin ''
+    # Replace hardcoded path to strip with llvm-strip
+    # https://github.com/NixOS/nixpkgs/issues/299606
+    substituteInPlace compiler/rustc_codegen_ssa/src/back/link.rs \
+      --replace-fail \
+        'let stripcmd = if cfg!(target_os = "macos") { "/usr/bin/strip" } else { "strip" };' \
+        'let stripcmd = "${lib.getExe' llvmPackages.llvm "llvm-strip"}";'
```

Comment:
Shouldn't this be llvmShared, so we don't end up referencing multiple LLVMs?

Reply:
could we punt on understanding the LLVMs and just use `${lib.getExe' stdenv.cc "strip"}` ?

Reply:
@reckenrode ?

Reply:
> could we punt on understanding the LLVMs and just use `${lib.getExe' stdenv.cc "strip"}` ?

eh, maybe that is even worse due to the symlinks. disregard.